### ISSUE TITLE
Do not throw `_Map_base::at` when receiving an event for a recently deleted watchpoint

### ISSUE THREAD

**Issue by wolfs** [gradle/native-platform#194]
Under some circumstances, the Linux file watcher may receive a change event which is not `IN_IGNORED` for a recently removed watched point. The native side then throws an `out_of_range` exception when trying to access the path for the watch descriptor here: https://github.com/gradle/native-platform/blob/master/src/file-events/cpp/linux_fsnotifier.cpp#L184

See this [build scan](https://e.grdev.net/s/rimrty7xuumvm/tests/ykfkc5kxrsfs4-u6vxcjrarml7u). `SourceDependencyCleanupIntegrationTest` seems to expose the problem on the Gradle side.

**Comment by lptr** [comment]
I don't understand how this has not happened before. We've done a ton of stress testing etc. What changed?

**Comment by wolfs** [comment]
I think there are two reasons we didn't see this before:
- The registering/unregistering and the handling of events was happening on the same thread, so while there were events those were handled first until we came around handling the events for registering/unregistering. Now it is possible that the unregister command comes in while we actually have events and the watch descriptor has already been closed.
- In the stress tests, we don't test unregistering a watched path during the test, we only test closing the watcher.
